### PR TITLE
Report all akka queue errors properly

### PR DESCRIPTION
**Pull request by tomrf1**
This was a TODO that I forgot to do...
The app uses a queue for sending messages to facebook. `queue.offer` returns a result based on the `QueueOfferResult` sealed trait.
I've noticed a couple of error logs caused by the queue rejecting messages. This change will help clarify what the cause is, if it happens again.

### CODE REVIEW

**Comment by shtukas** [comment]
:+1:

File: src/main/scala/com/gu/facebook_news_bot/services/Facebook.scala [hunk 1]
```diff
@@ -149,7 +150,8 @@ class FacebookImpl extends Facebook with CirceSupport {
       queue.offer(request -> promise).flatMap {
         case QueueOfferResult.Enqueued => promise.future
         case QueueOfferResult.Failure(e) => Future.failed(e)
-        case _ => Future.failed(new RuntimeException())
+        case QueueOfferResult.Dropped => Future.failed(EnqueueException(QueueOfferResult.Dropped))
+        case QueueOfferResult.QueueClosed => Future.failed(EnqueueException(QueueOfferResult.QueueClosed))
```
Comment (LATaylor-guardian) [inline src/main/scala/com/gu/facebook_news_bot/services/Facebook.scala]: Is this exhaustive?

Comment (shtukas) [inline src/main/scala/com/gu/facebook_news_bot/services/Facebook.scala]: Sealed trait. I asked the same question :)

**Comment by LATaylor-guardian** [comment]
👍

File: src/main/scala/com/gu/facebook_news_bot/services/Facebook.scala (hunk 1 above)
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/services/Facebook.scala]: yes - http://doc.akka.io/japi/akka/2.4/akka/stream/QueueOfferResult.html

**Pull request merged**